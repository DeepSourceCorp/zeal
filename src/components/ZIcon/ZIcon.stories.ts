import ZIcon from './ZIcon.vue'

import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'

export default {
  title: 'Icons',
  component: ZIcon,
  excludeStories: /.*Data$/
}

export const Default = () => ({
  components: { ZIcon },
  template: `<div class='wrapper'>
        <z-icon icon='arrow-up'></z-icon>
        <p class="icon-title">arrow-up</p>
    </div>`
})

export const IconWithCustomColor = () => ({
  components: { ZIcon },
  template: `<div class="wrapper">
        <z-icon icon="activity" color="juniper"></z-icon>
        <p class="icon-title">activity</p>
    </div>`
})

export const SmallIcon = () => ({
  components: { ZIcon },
  template: `<div class='wrapper'>
        <z-icon icon='arrow-up' size="small"></z-icon>
        <p class='icon-title'>arrow-up</p>
    </div>`
})
