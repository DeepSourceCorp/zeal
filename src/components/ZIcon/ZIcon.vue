<template>
    <svg 
      :class="[
        'z-icon',
        `${size && `z-icon--${size}`}`,
        `${icon && `z-icon--${icon}`}`
      ]"
      viewBox="0 0 24 24" 
      v-html="getIcon(icon)"></svg>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import feather from 'feather-icons'
import customIcons from '../../utils/icon-helper'

@Component({
  name: 'ZIcon'
})
export default class Icon extends Vue {
  @Prop({ default: '' })
  private icon!: string
  @Prop({default: ''})
  private size!: string

  public getIcon(iconName: string): HTMLOrSVGImageElement {
    let DOM = null
    if (feather.icons[iconName]) {
      DOM = feather.icons[iconName] && feather.icons[iconName].contents
    } else {
      DOM = customIcons[iconName] &&  customIcons[iconName].contents;
    }
    return DOM
  }
}
</script>

<style lang="css" scoped>
  .z-icon {
    @apply  w-6 h-6 fill-current stroke-current text-slate stroke-2;
  }

  .z-icon--small {
    @apply w-4 h-4;
  }
</style>
