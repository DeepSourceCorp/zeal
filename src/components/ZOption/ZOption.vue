<template>
  <div
    class="z-option p-2 cursor-pointer text-vanilla-300 hover:bg-ink-200 text-xs"
    :class="selected && 'is-selected bg-ink-200'"
    @click.stop="selectOption()"
  >
    <slot v-if="$slots.default"></slot>
    <template v-else>{{ label || value }}</template>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

interface ZSelectT extends Vue {
  selectedOpt: string | number
  selectedOptLabel: string | number
  open: boolean
}

export default Vue.extend({
  name: 'ZOption',
  props: {
    label: {
      type: String
    },
    value: {
      type: String || Number
    }
  },
  computed: {
    selected(): boolean {
      const $parent = this.$parent as ZSelectT
      return this.value === $parent.selectedOpt
    }
  },
  methods: {
    selectOption(): void {
      const $parent = this.$parent as ZSelectT
      $parent.selectedOpt = this.value
      $parent.selectedOptLabel = this.label || this.value
      $parent.open = false
    }
  }
})
</script>
