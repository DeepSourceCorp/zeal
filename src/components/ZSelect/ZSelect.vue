<template>
  <div
    :tabindex="tabIndex"
    class="relative w-full h-full leading-8 text-left outline-none custom-select"
    :class="{ 'is-disabled': disabled, 'is-readonly': readOnly }"
    @blur.stop="!disabled && !readOnly && blurEvent()"
  >
    <div
      class="flex items-center justify-between h-full space-x-2 border border-solid selected"
      :class="[
        (open && 'border-vanilla-400') || borderClass,
        spacing,
        backgroundClass,
        borderRadius,
        getTextColor,
        getCursorType
      ]"
      @[canClick]="open = !open"
    >
      <slot name="icon"></slot>
      <div
        class="flex items-center flex-grow h-full outline-none cursor-pointer"
        :class="[
          getTextSize,
          getCursorType,
          selectedValue === undefined ? 'text-vanilla-400 opacity-70' : '',
          { truncate }
        ]"
      >
        <template v-if="selectedLabel || selectedValue">
          {{ selectedLabel || selectedValue }}
        </template>
        <template v-else>
          {{ placeholder }}
        </template>
      </div>
      <button v-if="selectedValue && clearable" class="flex items-center justify-between" @click.stop="clearSelected">
        <z-icon icon="x" size="small" :color="getIconColor"></z-icon>
      </button>
      <span v-else>
        <z-icon
          icon="chevron-down"
          size="small"
          class="transition-all duration-300 transform"
          :color="getIconColor"
          :class="(open && 'rotate-180') || 'rotate-0'"
        ></z-icon>
      </span>
    </div>
    <!-- prettier-ignore -->
    <div
      class="absolute left-0 right-0 z-10 mt-1 overflow-y-auto transition-all duration-300 border border-solid rounded-md max-h-64 options shadow-black border-ink-100 text-vanilla-300 bg-ink-300"
      :class="!open && 'hidden'"
    >
      <slot></slot>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import ZIcon from '@/components/ZIcon/ZIcon.vue'

interface ZOptionPropsT extends Vue {
  value: string | number | null
  label: string
}

export default Vue.extend({
  name: 'ZSelect',
  components: {
    ZIcon
  },
  props: {
    selected: {
      type: [String, Number],
      default: null
    },
    tabIndex: {
      type: Number,
      required: false,
      default: 0
    },
    placeholder: {
      type: String,
      default: 'Select an Option'
    },
    clearable: {
      type: Boolean,
      default: false
    },
    spacing: {
      type: String,
      default: 'px-2 py-3'
    },
    borderClass: {
      type: String,
      default: 'border-ink-200'
    },
    borderRadius: {
      type: String,
      default: 'rounded-md'
    },
    backgroundClass: {
      type: String,
      default: 'bg-transparent'
    },
    textSize: {
      type: String,
      default: 'text-xs'
    },
    disabled: {
      default: false,
      type: Boolean
    },
    readOnly: {
      default: false,
      type: Boolean
    },
    truncate: {
      default: false,
      type: Boolean
    }
  },

  model: {
    prop: 'selected',
    event: 'change'
  },

  data() {
    return {
      open: false,
      options: [] as Array<Vue>
    }
  },

  computed: {
    canClick(): string {
      if (!this.disabled && !this.readOnly) return 'click'
      return ''
    },
    getTextSize(): string {
      const validTextSizes = ['text-xs', 'text-sm', 'text-base', 'text-lg', 'text-xl']
      if (validTextSizes.includes(this.textSize)) {
        return this.textSize
      }
      return 'text-xs'
    },
    getTextColor(): string {
      if (this.disabled) return 'text-slate'
      if (this.readOnly) return 'text-vanilla-400'
      return 'text-vanilla-300'
    },
    getCursorType(): string {
      if (this.disabled || this.readOnly) return 'cursor-not-allowed'
      return 'cursor-pointer'
    },
    getIconColor(): string {
      if (this.disabled) return 'slate'
      return 'vanilla-400'
    },
    selectedLabel(): string | number | null {
      const selectedOption = this.getSelectedOption()

      if (selectedOption) {
        return selectedOption.label
      }
      return null
    },
    selectedValue(): string | number | null {
      const selectedOption = this.getSelectedOption()

      if (selectedOption) {
        return selectedOption.value
      }
      return null
    }
  },

  mounted(): void {
    this.options = this.$children.filter((child) => child.$options.name === 'ZOption')
  },

  methods: {
    blurEvent(): void {
      this.open = false
    },
    clearSelected(): void {
      this.$emit('change', null)
    },
    getSelectedOption(): ZOptionPropsT | null {
      if (this.selected) {
        const selectedOption = this.options
          .map((child) => {
            return child.$options.propsData as ZOptionPropsT
          })
          .find((childProp) => {
            return childProp.value === this.selected
          })

        if (selectedOption) {
          return selectedOption
        }
      }

      return null
    },
    onChange(val: string): void {
      this.$emit('change', val)
    }
  }
})
</script>
