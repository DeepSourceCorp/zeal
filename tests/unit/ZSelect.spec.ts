import ZSelect from '../../src/components/ZSelect'
import ZOption from '../../src/components/ZOption'
import ZIcon from '../../src/components/ZIcon'

import { mount } from '@vue/test-utils'

const defaultSelect = {
  template: `<z-select v-model="value">
                <z-option
                    v-for="item in options"
                    :key="item.value"
                    :label="item.label"
                    :value="item.value">
                </z-option>
            </z-select>`,
  components: {
    ZSelect,
    ZOption
  },
  data() {
    return {
      value: '',
      options: [
        {
          value: 'Option 1',
          label: 'Option 1'
        },
        {
          value: 'Option 2',
          label: 'Option 2'
        }
      ]
    }
  }
}

const SelectWithCustomPlaceholder = {
  template: `<z-select v-model="value" placeholder="Choose a framework">
                <z-option
                    v-for="item in options"
                    :key="item.value"
                    :label="item.label"
                    :value="item.value">
                </z-option>
            </z-select>`,
  components: {
    ZSelect,
    ZOption
  },
  data() {
    return {
      value: '',
      options: [
        {
          value: 'Option 1',
          label: 'Option 1'
        },
        {
          value: 'Option 2',
          label: 'Option 2'
        }
      ]
    }
  }
}

const SelectWithClearables = {
  template: `<z-select v-model="value" placeholder="Choose a framework" :clearable="true" selected="Option 1">
                <z-option
                    v-for="item in options"
                    :key="item.value"
                    :label="item.label"
                    :value="item.value">
                </z-option>
            </z-select>`,
  components: {
    ZSelect,
    ZOption
  },
  data() {
    return {
      value: '',
      options: [
        {
          value: 'Option 1',
          label: 'Option 1'
        },
        {
          value: 'Option 2',
          label: 'Option 2'
        }
      ]
    }
  }
}

const SelectWithCustomOptionsTemplate = {
  template: `<z-select v-model="value" placeholder="Choose a framework" :clearable="true">
                    <z-option
                        v-for="item in options"
                        :key="item.value"
                        :label="item.label"
                        :value="item.value">
                        <div class="flex items-center">
                            <z-icon icon="github" size="small" class="-mt-1 mr-2"></z-icon>
                            <span class="flex-1">{{item.label}}</span>
                        </div>
                    </z-option>
                </z-select>`,
  components: {
    ZSelect,
    ZOption,
    ZIcon
  },
  data() {
    return {
      value: 'Option 1',
      options: [
        {
          value: 'Option 1',
          label: 'Option 1'
        },
        {
          value: 'Option 2',
          label: 'Option 2'
        }
      ]
    }
  }
}

const disabledSelect = {
  template: `<z-select v-model="value" :disabled="true">
                <z-option
                    v-for="item in options"
                    :key="item.value"
                    :label="item.label"
                    :value="item.value">
                </z-option>
            </z-select>`,
  components: {
    ZSelect,
    ZOption
  },
  data() {
    return {
      value: '',
      options: [
        {
          value: 'Option 1',
          label: 'Option 1'
        },
        {
          value: 'Option 2',
          label: 'Option 2'
        }
      ]
    }
  }
}

const readOnlySelect = {
  template: `<z-select v-model="value" :read-only="true">
                <z-option
                    v-for="item in options"
                    :key="item.value"
                    :label="item.label"
                    :value="item.value">
                </z-option>
            </z-select>`,
  components: {
    ZSelect,
    ZOption
  },
  data() {
    return {
      value: '',
      options: [
        {
          value: 'Option 1',
          label: 'Option 1'
        },
        {
          value: 'Option 2',
          label: 'Option 2'
        }
      ]
    }
  }
}

describe('Select Component', () => {
  it('renders the default select component', () => {
    const wrapper = mount(defaultSelect)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('.z-option').length).toEqual(2)
  })

  it('renders the select with a custom placeholder', () => {
    const wrapper = mount(SelectWithCustomPlaceholder)
    expect(wrapper.html()).toMatchSnapshot()
    wrapper.find('.is-selected div')
  })
  it('renders the select component with clearables', () => {
    const wrapper = mount(SelectWithClearables)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('.z-icon').length).toEqual(1)
  })
  it('renders a select component with custom options template', () => {
    const wrapper = mount(SelectWithCustomOptionsTemplate)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('.z-option svg').length).toEqual(2)
  })
  it('renders a disabled select component', () => {
    const wrapper = mount(disabledSelect)
    expect(wrapper.classes()).toContain('is-disabled')
    expect(wrapper.findAll('.z-option').length).toEqual(2)
  })
  it('renders a read only select component', () => {
    const wrapper = mount(readOnlySelect)
    expect(wrapper.classes()).toContain('is-readonly')
    expect(wrapper.findAll('.z-option').length).toEqual(2)
  })
})
