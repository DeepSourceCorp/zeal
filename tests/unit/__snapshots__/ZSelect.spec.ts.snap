// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component renders a select component with custom options template 1`] = `
<div tabindex="0" class="relative w-full h-full leading-8 text-left outline-none custom-select">
  <div class="flex items-center justify-between h-full space-x-2 border border-solid selected border-ink-200 px-2 py-3 bg-transparent rounded-md text-vanilla-300 cursor-pointer">
    <div class="flex items-center flex-grow h-full outline-none cursor-pointer text-xs cursor-pointer text-vanilla-400 opacity-70">
      Choose a framework
    </div> <span><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="transition-all duration-300 transform z-icon w-4 h-4 z-icon--chevron-down text-vanilla-400 rotate-0"><polyline points="6 9 12 15 18 9"></polyline></svg></span>
  </div>
  <div class="absolute left-0 right-0 z-10 mt-1 overflow-y-auto transition-all duration-300 border border-solid rounded-md options shadow-black border-ink-100 text-vanilla-300 bg-ink-300 hidden max-h-64">
    <div class="p-2 cursor-pointer z-option text-vanilla-300 hover:bg-ink-200 is-selected bg-ink-200 text-xs">
      <div class="flex items-center"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="-mt-1 mr-2 z-icon w-4 h-4 z-icon--github text-vanilla-400">
          <path d="M11.9947 2.24707C9.62096 2.24717 7.32472 3.0919 5.51684 4.63011C3.70896 6.16832 2.5074 8.29963 2.12716 10.6427C1.74692 12.9858 2.21282 15.3877 3.44148 17.4187C4.67015 19.4497 6.5814 20.9772 8.83327 21.7279C9.3331 21.8154 9.52053 21.5155 9.52053 21.2531C9.52053 21.0157 9.50803 20.2284 9.50803 19.3912C6.99637 19.8536 6.34659 18.779 6.14666 18.2167C5.92481 17.6698 5.57313 17.1851 5.12201 16.8046C4.77213 16.6172 4.2723 16.1548 5.10951 16.1424C5.42918 16.1771 5.73579 16.2883 6.00334 16.4667C6.2709 16.645 6.49151 16.8852 6.64649 17.167C6.78322 17.4126 6.96706 17.6288 7.1875 17.8033C7.40793 17.9777 7.66062 18.1069 7.93109 18.1835C8.20155 18.2602 8.48448 18.2827 8.76365 18.2497C9.04282 18.2168 9.31276 18.1291 9.55798 17.9917C9.60126 17.4835 9.82775 17.0083 10.1953 16.6547C7.97104 16.4048 5.64683 15.5425 5.64683 11.7189C5.63279 10.7254 5.9994 9.76412 6.67148 9.03229C6.36587 8.16881 6.40163 7.22119 6.77145 6.38319C6.77145 6.38319 7.60863 6.12077 9.52051 7.40784C11.1562 6.95797 12.8831 6.95797 14.5188 7.40784C16.4306 6.10829 17.2679 6.38319 17.2679 6.38319C17.6378 7.22118 17.6735 8.16881 17.3678 9.03229C18.042 9.76287 18.4089 10.7249 18.3925 11.7189C18.3925 15.555 16.0558 16.4048 13.8315 16.6547C14.0701 16.8965 14.2539 17.1868 14.3703 17.5059C14.4867 17.825 14.5331 18.1654 14.5063 18.5041C14.5063 19.8411 14.4938 20.9157 14.4938 21.2531C14.4938 21.5156 14.6812 21.8279 15.1811 21.728C17.429 20.9711 19.3347 19.44 20.5582 17.408C21.7816 15.3759 22.2431 12.9752 21.8602 10.6345C21.4773 8.29366 20.275 6.16511 18.4679 4.62877C16.6608 3.09243 14.3666 2.2483 11.9947 2.24707Z" fill="#C0C1C3"></path>
        </svg> <span class="flex-1">Option 1</span></div>
    </div>
    <div class="p-2 cursor-pointer z-option text-vanilla-300 hover:bg-ink-200 text-xs">
      <div class="flex items-center"><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="-mt-1 mr-2 z-icon w-4 h-4 z-icon--github text-vanilla-400">
          <path d="M11.9947 2.24707C9.62096 2.24717 7.32472 3.0919 5.51684 4.63011C3.70896 6.16832 2.5074 8.29963 2.12716 10.6427C1.74692 12.9858 2.21282 15.3877 3.44148 17.4187C4.67015 19.4497 6.5814 20.9772 8.83327 21.7279C9.3331 21.8154 9.52053 21.5155 9.52053 21.2531C9.52053 21.0157 9.50803 20.2284 9.50803 19.3912C6.99637 19.8536 6.34659 18.779 6.14666 18.2167C5.92481 17.6698 5.57313 17.1851 5.12201 16.8046C4.77213 16.6172 4.2723 16.1548 5.10951 16.1424C5.42918 16.1771 5.73579 16.2883 6.00334 16.4667C6.2709 16.645 6.49151 16.8852 6.64649 17.167C6.78322 17.4126 6.96706 17.6288 7.1875 17.8033C7.40793 17.9777 7.66062 18.1069 7.93109 18.1835C8.20155 18.2602 8.48448 18.2827 8.76365 18.2497C9.04282 18.2168 9.31276 18.1291 9.55798 17.9917C9.60126 17.4835 9.82775 17.0083 10.1953 16.6547C7.97104 16.4048 5.64683 15.5425 5.64683 11.7189C5.63279 10.7254 5.9994 9.76412 6.67148 9.03229C6.36587 8.16881 6.40163 7.22119 6.77145 6.38319C6.77145 6.38319 7.60863 6.12077 9.52051 7.40784C11.1562 6.95797 12.8831 6.95797 14.5188 7.40784C16.4306 6.10829 17.2679 6.38319 17.2679 6.38319C17.6378 7.22118 17.6735 8.16881 17.3678 9.03229C18.042 9.76287 18.4089 10.7249 18.3925 11.7189C18.3925 15.555 16.0558 16.4048 13.8315 16.6547C14.0701 16.8965 14.2539 17.1868 14.3703 17.5059C14.4867 17.825 14.5331 18.1654 14.5063 18.5041C14.5063 19.8411 14.4938 20.9157 14.4938 21.2531C14.4938 21.5156 14.6812 21.8279 15.1811 21.728C17.429 20.9711 19.3347 19.44 20.5582 17.408C21.7816 15.3759 22.2431 12.9752 21.8602 10.6345C21.4773 8.29366 20.275 6.16511 18.4679 4.62877C16.6608 3.09243 14.3666 2.2483 11.9947 2.24707Z" fill="#C0C1C3"></path>
        </svg> <span class="flex-1">Option 2</span></div>
    </div>
  </div>
</div>
`;

exports[`Select Component renders a truncated select component 1`] = `
<div class="padded-container">
  <div class="select-container">
    <div tabindex="0" class="relative w-full h-full leading-8 text-left outline-none custom-select text-sm">
      <div class="flex items-center justify-between h-full space-x-2 border border-solid selected border-ink-200 px-4 py-3 bg-transparent rounded-md text-vanilla-300 cursor-pointer">
        <div class="flex items-center flex-grow h-full outline-none cursor-pointer text-xs cursor-pointer text-vanilla-400 opacity-70 truncate">
          Select an Option
        </div> <span><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="transition-all duration-300 transform z-icon w-4 h-4 z-icon--chevron-down text-vanilla-400 rotate-0"><polyline points="6 9 12 15 18 9"></polyline></svg></span>
      </div>
      <div class="absolute left-0 right-0 z-10 mt-1 overflow-y-auto transition-all duration-300 border border-solid rounded-md options shadow-black border-ink-100 text-vanilla-300 bg-ink-300 hidden max-h-64">
        <div class="p-2 cursor-pointer z-option text-vanilla-300 hover:bg-ink-200 px-4 overflow-x-scroll text-xs">Option 1</div>
        <div class="p-2 cursor-pointer z-option text-vanilla-300 hover:bg-ink-200 px-4 overflow-x-scroll text-xs">Adipisicing fugiat minim consectetur labore nulla quis sunt quis amet anim</div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Select Component renders the default select component 1`] = `
<div tabindex="0" class="relative w-full h-full leading-8 text-left outline-none custom-select">
  <div class="flex items-center justify-between h-full space-x-2 border border-solid selected border-ink-200 px-2 py-3 bg-transparent rounded-md text-vanilla-300 cursor-pointer">
    <div class="flex items-center flex-grow h-full outline-none cursor-pointer text-xs cursor-pointer text-vanilla-400 opacity-70">
      Select an Option
    </div> <span><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="transition-all duration-300 transform z-icon w-4 h-4 z-icon--chevron-down text-vanilla-400 rotate-0"><polyline points="6 9 12 15 18 9"></polyline></svg></span>
  </div>
  <div class="absolute left-0 right-0 z-10 mt-1 overflow-y-auto transition-all duration-300 border border-solid rounded-md options shadow-black border-ink-100 text-vanilla-300 bg-ink-300 hidden max-h-64">
    <div class="p-2 cursor-pointer z-option text-vanilla-300 hover:bg-ink-200 text-xs">Option 1</div>
    <div class="p-2 cursor-pointer z-option text-vanilla-300 hover:bg-ink-200 text-xs">Option 2</div>
  </div>
</div>
`;

exports[`Select Component renders the select component with clearables 1`] = `
<div tabindex="0" class="relative w-full h-full leading-8 text-left outline-none custom-select">
  <div class="flex items-center justify-between h-full space-x-2 border border-solid selected border-ink-200 px-2 py-3 bg-transparent rounded-md text-vanilla-300 cursor-pointer">
    <div class="flex items-center flex-grow h-full outline-none cursor-pointer text-xs cursor-pointer text-vanilla-400 opacity-70">
      Choose a framework
    </div> <span><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="transition-all duration-300 transform z-icon w-4 h-4 z-icon--chevron-down text-vanilla-400 rotate-0"><polyline points="6 9 12 15 18 9"></polyline></svg></span>
  </div>
  <div class="absolute left-0 right-0 z-10 mt-1 overflow-y-auto transition-all duration-300 border border-solid rounded-md options shadow-black border-ink-100 text-vanilla-300 bg-ink-300 hidden max-h-64">
    <div class="p-2 cursor-pointer z-option text-vanilla-300 hover:bg-ink-200 text-xs">Option 1</div>
    <div class="p-2 cursor-pointer z-option text-vanilla-300 hover:bg-ink-200 text-xs">Option 2</div>
  </div>
</div>
`;

exports[`Select Component renders the select with a custom placeholder 1`] = `
<div tabindex="0" class="relative w-full h-full leading-8 text-left outline-none custom-select">
  <div class="flex items-center justify-between h-full space-x-2 border border-solid selected border-ink-200 px-2 py-3 bg-transparent rounded-md text-vanilla-300 cursor-pointer">
    <div class="flex items-center flex-grow h-full outline-none cursor-pointer text-xs cursor-pointer text-vanilla-400 opacity-70">
      Choose a framework
    </div> <span><svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="transition-all duration-300 transform z-icon w-4 h-4 z-icon--chevron-down text-vanilla-400 rotate-0"><polyline points="6 9 12 15 18 9"></polyline></svg></span>
  </div>
  <div class="absolute left-0 right-0 z-10 mt-1 overflow-y-auto transition-all duration-300 border border-solid rounded-md options shadow-black border-ink-100 text-vanilla-300 bg-ink-300 hidden max-h-64">
    <div class="p-2 cursor-pointer z-option text-vanilla-300 hover:bg-ink-200 text-xs">Option 1</div>
    <div class="p-2 cursor-pointer z-option text-vanilla-300 hover:bg-ink-200 text-xs">Option 2</div>
  </div>
</div>
`;
